src/binder/clrprivbindercoreclr.cpp

```diff
@@ -157,10 +149,12 @@ HRESULT CLRPrivBinderCoreCLR::BindUsingPEImage( /* in */ PEImage *pPEImage,
                 hr = BindAssemblyByNameWorker(pAssemblyName, &pCoreCLRFoundAssembly, true /* excludeAppPaths */);
                 if (SUCCEEDED(hr))
                 {
+                    // Is identity same as TPA assembly
                     if (pCoreCLRFoundAssembly->GetIsInGAC())
                     {
-                        // If we were able to bind to a TPA assembly, then fail the load
-                        IF_FAIL_GO(HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND));
+                        // Fail bind if requested assembly is at different location than TPA assembly
+                        if(pPEImage != pCoreCLRFoundAssembly->GetPEImage())
```

Comment:
Yes. I am not very strongly against doing what you are suggesting ...only rationale I had was to be explicit and cause an error instead of loading dll at some other location. But we already do that for ALC.LoadFromAssemblyPath wherein we return the already loaded assembly instead of an error. I will do the change as you are suggesting. @jkotas do you have any objection?